test/src/test/java/hudson/slaves/SlaveComputerTest.java

```diff
@@ -228,4 +231,51 @@ private String getRemoteFS(Node node, String user) throws Exception {
             return pathObj.toString();
         }
     }
+
+    @Test
+    void testAgentSecretWithAgentConnectPermission() throws Exception {
+        DumbSlave testAgent = j.createOnlineSlave();
+
+        // Setup user with Agent/Connect permission
+        String userWithConnect = "user-with-connect";
+        MockAuthorizationStrategy authStrategy = new MockAuthorizationStrategy();
+        authStrategy.grant(Computer.CONNECT, Jenkins.READ).everywhere().to(userWithConnect);
+
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
+        j.jenkins.setAuthorizationStrategy(authStrategy);
+
+        JenkinsRule.WebClient wc = j.createWebClient();
+        wc.login(userWithConnect);
+
+        Page page = wc.goTo("computer/" + testAgent.getNodeName() + "/agent-secret", "text/plain");
+        WebResponse response = page.getWebResponse();
+
+        // Verify response
+        assertEquals(200, response.getStatusCode());
+
+        String secret = response.getContentAsString().trim();
+        assertNotNull(secret);
+        assertEquals(testAgent.getComputer().getJnlpMac(), secret);
+    }
+
+    @Test
+    void testAgentSecretWithoutAgentConnectPermission() throws Exception {
+        DumbSlave testAgent = j.createOnlineSlave();
+
+        // Setup user without Agent/Connect permission
+        String userWithoutConnect = "user-without-connect";
+        MockAuthorizationStrategy authStrategy = new MockAuthorizationStrategy();
+
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
```

Comment:
You should still grant the user  the permission `Jenkins.READ`. Otherwise you don't test if your method really performs the permission check as it already fails when accessing the computer before it comes to executing your new method. Verify that by debugging the code